### PR TITLE
fix-3091 added simple mapping for constatnt expressions

### PR DESCRIPTION
**Pull request by bhavnajindal**
Issue: https://github.ibm.com/ClickHouse/issue-repo/issues/3091

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Requirement was to add a direct mapping without using expression inside the array. Therefore, now computing the index first and then passing it to CH SQL only for constant numerical expressions.

Query:
`select * from kql(events | project custom_prop_value[0] | take 1)`

Output:
```
SELECT *
FROM
(
    SELECT custom_prop_value[1] AS Column1
    FROM events
    LIMIT 1
)
```

### CODE REVIEW

**Review by kashwy — COMMENTED** [review]
add some test cases of negative index

File: src/Parsers/tests/KQL/gtest_KQL_dynamicFunctions.cpp [hunk 1]
```diff
@@ -103,31 +103,31 @@ INSTANTIATE_TEST_SUITE_P(ParserKQLQuery_Dynamic, ParserTest,
...
         },
         {
             "print dynamic([[1,2,3,4,5],[20,30]])[1][1]",
-            "SELECT ([[1, 2, 3, 4, 5], [20, 30]][if(1 >= 0, 1 + 1, 1)])[if(1 >= 0, 1 + 1, 1)] AS print_0"
+            "SELECT ([[1, 2, 3, 4, 5], [20, 30]][2])[2] AS print_0"
         },
         {
             "print A[B[1]]",
-            "SELECT A[if((B[if(1 >= 0, 1 + 1, 1)]) >= 0, (B[if(1 >= 0, 1 + 1, 1)]) + 1, B[if(1 >= 0, 1 + 1, 1)])] AS print_0"
+            "SELECT A[if((B[2]) >= 0, (B[2]) + 1, B[2])] AS print_0"
```
Comment (kashwy) [inline src/Parsers/tests/KQL/gtest_KQL_dynamicFunctions.cpp:122]: can this recursive be solved too?

Comment (kashwy) [reply src/Parsers/tests/KQL/gtest_KQL_dynamicFunctions.cpp:134]: > We would need to get the value of B[2] to correctly map it, which isn't available during Parsing. Therefore, treating B[2] as an arbitrary expression.

ok, that's good

**Review by kashwy — APPROVED** [review]
